receiver/prometheusreceiver/factory.go

```diff
@@ -54,5 +55,9 @@ func createMetricsReceiver(
 	cfg component.Config,
 	nextConsumer consumer.Metrics,
 ) (component.MetricsReceiver, error) {
-	return newPrometheusReceiver(set, cfg.(*Config), nextConsumer, featuregate.GetRegistry()), nil
+	c := cfg.(*Config)
+	if c.PrometheusConfig == nil {
```

Comment:
should we do this in createDefaultConfig instead?  That way we still get global defaults even if the config itself is set.

Reply:
thanks for your review, @dashpole . I made a change to move this to `createDefaultConfig()` .